zulip_bots/zulip_bots/bots/github_detail/github_detail.py

```diff
@@ -74,22 +74,27 @@ def handle_message(self, message: Dict[str, str], bot_handler: Any) -> None:
             return
 
         # Capture owner, repo, id
-        issue_prs = re.finditer(
-            self.HANDLE_MESSAGE_REGEX, message['content'])
+        issue_prs = list(re.finditer(
+            self.HANDLE_MESSAGE_REGEX, message['content']))
         bot_messages = []
-        for issue_pr in issue_prs:
-            owner, repo = self.get_owner_and_repo(issue_pr)
-            if owner and repo:
-                details = self.get_details_from_github(owner, repo, issue_pr.group(3))
-                if details is not None:
-                    details['owner'] = owner
-                    details['repo'] = repo
-                    bot_messages.append(self.format_message(details))
+        # issue_prs = list(issue_prs)
+        if len(issue_prs) > 5:
+            bot_messages.append('Please ask for <=5 links in any one request')
+            # We limit to 5 requests to prevent denial-of-service
```

Comment:
I would just do an early `return` here.  We generally don't like to put the main logic in an `else` statement.